lib/LaTeXML/Package.pm

```diff
@@ -2846,24 +2847,48 @@ sub decodeMathChar {
     $fontdef   = T_CS('\font');    # Assume specified by \mathrm or something similar!
     $fontinfo  = $STATE->lookupValue('font')->asFontinfo; }
   else {
-    $fontdef = LookupValue('textfont_' . $fam);
+    my $style = $curfont->getMathstyle;
+    $style = 'text' unless $style && ($style =~ /^(:?scriptscript|script|text)$/);
+    my $basefontdef  = LookupValue('textfont_0');
+    my $basefontdefn = $STATE->lookupDefinition($basefontdef);
+    my $basefontinfo = $basefontdefn && $basefontdefn->isFontDef;
+    if ($style eq 'text') { # Lookup the requested font according to script level, but with adjusted fallbacks
+      $fontdef = LookupValue('textfont_' . $fam); }
+    elsif ($style eq 'script') {
+      if ($fontdef = LookupValue('scriptfont_' . $fam)) { }
+      elsif ($fontdef = LookupValue('textfont_' . $fam)) { $downsize = 1; } }
+    elsif ($style eq 'scriptscript') {
+      if    ($fontdef = LookupValue('scriptscriptfont_' . $fam)) { }
+      elsif ($fontdef = LookupValue('scriptfont_' . $fam))       { $downsize = 1; }
+      elsif ($fontdef = LookupValue('textfont_' . $fam))         { $downsize = 2; } }
     my $defn = $STATE->lookupDefinition($fontdef);
-    $fontinfo = $defn && $defn->isFontDef; }
-  my $font     = $curfont->merge(%$fontinfo);
+    $fontinfo = $defn && $defn->isFontDef;
+    if ($$basefontinfo{size} != $curfont->getSize) { # If we've gotten an explicit font SIZE change; Adjust!
+      $fontinfo = {%$fontinfo}; $$fontinfo{size} = $curfont->getSize; } }
```

Comment:
There is a small theoretical chance `$fontinfo` is `undef` here, so maybe default it to `{}` ?

```perl
 $fontinfo = ($defn && $defn->isFontDef) || {};
```

Reply:
thanks